source/common/router/config_impl.cc

```diff
@@ -1001,9 +1001,22 @@ void RouteEntryImplBase::validateClusters(
   }
 }
 
-const RouteSpecificFilterConfig*
-RouteEntryImplBase::perFilterConfig(const std::string& name) const {
-  return per_filter_configs_.get(name);
+void RouteEntryImplBase::traversePerFilterConfig(
+    const std::string& filter_name,
+    std::function<void(const Router::RouteSpecificFilterConfig&)> cb) const {
+  const Router::RouteEntry* route_entry = routeEntry();
```

Comment:
Isn't this the same bug? Shouldn't we not be gating this on `routeEntry()`? Or are you not fixing this because you want it changes extauth and you want to fix this separately? If the latter can you add a comment/TODO here?

Reply:
Yes, I plan to fix this in separate PR, along with https://github.com/envoyproxy/envoy/issues/17502, Let me add a todo.

Actually, I'm a little confused about the separate PR. I used to use gerrit before. Everybody is reviewing the each commit and merge each commit separately. But PR includes multiple commits, but look like people doesn't care about how many commits in a PR.  But let me know what is preference way people do here :)

Reply:
We squash all the commits when we merge, so just add as many commits as you want. The GH review system is very bad compared to Gerrit and this is the only way to make it reasonably work.